### PR TITLE
Support loading zoo checkpoint in TFOptimizer

### PR DESCRIPTION
**Pull request by yangw1234**
## Feature

Support loading zoo checkpoint in TFOptimizer and resume training

## Usage

```python
opt = TFOptimizer.from_loss(...)
opt.load_checkpoint(path, version=1000)
opt.optimize()
```

## Todos

- [x] unit tests
- [x] update inception v1 example
- [x] manually tested inception v1 example
- [x] jenkins test, http://10.239.47.210:18888/view/ZOO-PR/job/ZOO-PR-Validation/3228/, mxnet failure is irrelevant

### CODE REVIEW

File: pyzoo/zoo/tfpark/tf_optimizer.py [hunk 1]
```diff
@@ -395,6 +399,21 @@ def __init__(self, tf_model, optim_method,
             min_value, max_value = self.clip_constant
             self.estimator.set_constant_gradient_clipping(min_value, max_value)
 
+    def load_zoo_checkpoint(self, path, version):
```
Comment (jason-dai) [inline pyzoo/zoo/tfpark/tf_optimizer.py:402]: just `load_checkpoint` or `load_tfpark_checkpoint`

File: pyzoo/zoo/examples/tensorflow/tfpark/inception/inception.py [hunk 2]
```diff
@@ -56,6 +56,11 @@ def config_option_parser():
                       help="weight decay")
     parser.add_option("--checkpointIteration", type=int, dest="checkpointIteration", default=620,
                       help="checkpoint interval of iterations")
+    parser.add_option("--resumeTrainingCheckpoint", type=str, dest="resumeTrainingCheckpoint",
+                      default=None,
+                      help="a analytics zoo checkpoint path for resume training, usually contains"
```
Comment (jason-dai) [inline pyzoo/zoo/examples/tensorflow/tfpark/inception/inception.py:61]: an analytics zoo checkpoint path

File: pyzoo/zoo/tfpark/tf_optimizer.py [hunk 3]
```diff
@@ -91,6 +91,9 @@ def get_weights_to_python(self):
         self.save_checkpoint()
         self.saver.restore(self.sess, os.path.join(self.export_dir, "model"))
 
+    def load_zoo_checkpoint(self, path):
```
Comment (jason-dai) [inline pyzoo/zoo/tfpark/tf_optimizer.py:94]: just `load_checkpoint` or `load_tfpark_checkpoint`

File: pyzoo/zoo/tfpark/tf_optimizer.py (hunk 1 above)
Comment (yangw1234) [reply pyzoo/zoo/tfpark/tf_optimizer.py]: changed to load_checkpoint

File: pyzoo/zoo/examples/tensorflow/tfpark/inception/inception.py (hunk 2 above)
Comment (yangw1234) [reply pyzoo/zoo/examples/tensorflow/tfpark/inception/inception.py]: fixed

File: pyzoo/zoo/tfpark/tf_optimizer.py (hunk 3 above)
Comment (yangw1234) [reply pyzoo/zoo/tfpark/tf_optimizer.py]: changed to load_checkpoint

**Comment by yangw1234** [comment]
@hkvision mxnet example fails randomly, would you take a look? http://10.239.47.210:18888/job/ZOO-PR-Python-ExampleTests-py36/439/console

**Comment by hkvision** [comment]
Looking into this... weird...

**Pull request merged**